harness/determined/experimental/client.py

```diff
@@ -225,14 +226,21 @@ def whoami() -> User:
     return _determined.whoami()
 
 
-@_require_singleton
+# DOES NOT REQUIRE SINGLETON (don't force a login in order to log out).
 def logout() -> None:
     """
     Get the :class:`~determined.experimental.client.User` representing the
     current user.
     """
-    assert _determined is not None
-    return _determined.logout()
+    if _determined is not None:
+        return _determined.logout()
+
+    logging.warning(
+        "client has not been logged in, either explicitly by client.login() or implicitly by any "
+        "other client.* function, so client.logout() has no session to log out of and is a no-op. "
+        "If you would like to log out of the default active session, try "
+        "client.Determined().logout() instead."
+    )
```

Comment:
@stoksc just fyi I waffled again on what to do in a script that only calls `client.logout()`.  I decided it should actually be a noop, because otherwise for it to be meaningful `client.logout()` would have to accept `master_address`, `username`, and `cert` parameters, which `Determined.logout()` does not need or want.  But `client.Determined().logout()` would meet that user need just fine I think.